### PR TITLE
test(common/web/types): unit tests for string-list

### PR DESCRIPTION
**Pull request by markcsinclair**
Unit tests for `ldml-keyboard/string-list.ts'

See [#9052](https://github.com/keymanapp/keyman/issues/9052)

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by markcsinclair** [comment]
Hmm ... I have been happily typing `(developer)` on most of the commits ... oops!

**Comment by markcsinclair** [comment]
Ready for review ... as this has largely been done in a single day following a context switch from C++, please check carefully.

File: common/web/types/test/ldml-keyboard/test-string-list.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
+import 'mocha';
```
[the remark this answers is not in the document]
Comment (markcsinclair) [reply common/web/types/test/ldml-keyboard/test-string-list.ts:9]: Done

File: common/web/types/test/ldml-keyboard/test-string-list.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+});
+
+function stubSectionsStrsAllocString(s?: string, opts?: StrsOptions, sections?: DependencySections): StrsItem {
+  return new StrsItem(s);
+}
+
+function initListItem(source: Array<string>): ListItem {
+  const listItem = new ListItem();
+  for (const s of source)
+    listItem.push(new ListIndex(new StrsItem(s)));
```
[the remark this answers is not in the document]
Comment (markcsinclair) [reply common/web/types/test/ldml-keyboard/test-string-list.ts]: Done

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.147-alpha](https://keyman.com/downloads/releases/alpha/18.0.147)

**Pull request merged**